velox/functions/lib/SubscriptUtil.h

```diff
@@ -23,6 +23,22 @@
 
 namespace facebook::velox::functions {
 
+namespace {
+
+// A flat vector of map keys, an index into that vector and an index into
+// the original map keys vector that may have encodings.
+struct MapKey {
+  const BaseVector* baseVector;
+  const vector_size_t baseIndex;
+  const vector_size_t index;
+
+  static bool lessThen(const MapKey& left, const MapKey& right) {
```

Comment:
`lessThan`, or you can just overload `operator<` since it's a custom struct already